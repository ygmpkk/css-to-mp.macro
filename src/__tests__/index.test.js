const pluginTester = require('babel-plugin-tester')
const plugin = require('babel-plugin-macros')

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: {
    filename: __filename,
  },
  tests: {
    '转换CSS样式，一般模式': `
import css from '../macro'
const styles = css\`
  .container {
    flex: 1;
    justify-content: center;
    align-items: center;
    height: 100rpx;
  }
  .text {
    font-size: 12;
    font-weight: bold;
    color: #fefefe;
    padding-left: 20px;
  }
\`
    `,
    '转换CSS样式，带变量': `
import css from '../macro'

const height = 100;

const styles = css\`
  .container {
    flex: 1;
    justify-content: center;
    align-items: center;
    height: \${height};
  }
  .text {
    font-size: 12;
    font-weight: bold;
    color: #fefefe;
    padding-left: \${20 + height}px;
  }
\`
    `,

  },
})
