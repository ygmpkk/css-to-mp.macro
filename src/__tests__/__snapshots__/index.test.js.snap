// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros 转换CSS样式，一般模式: 转换CSS样式，一般模式 1`] = `
"
import css from '../macro'
const styles = css\`
  .container {
    flex: 1;
    justify-content: center;
    align-items: center;
    height: 100rpx;
  }
  .text {
    font-size: 12;
    font-weight: bold;
    color: #fefefe;
    padding-left: 20px;
  }
\`

      ↓ ↓ ↓ ↓ ↓ ↓

const styles = {
  'container': {
    'flex': 1,
    'justifyContent': 'center',
    'alignItems': 'center',
    'height': '100rpx'
  },
  'text': {
    'fontSize': 12,
    'fontWeight': 'bold',
    'color': '#fefefe',
    'paddingLeft': '20px'
  }
};
"
`;

exports[`macros 转换CSS样式，带变量: 转换CSS样式，带变量 1`] = `
"
import css from '../macro'

const height = 100;

const styles = css\`
  .container {
    flex: 1;
    justify-content: center;
    align-items: center;
    height: \${height};
  }
  .text {
    font-size: 12;
    font-weight: bold;
    color: #fefefe;
    padding-left: \${20 + height}px;
  }
\`

      ↓ ↓ ↓ ↓ ↓ ↓

const height = 100;

const styles = {
  'container': {
    'flex': 1,
    'justifyContent': 'center',
    'alignItems': 'center',
    'height': 100
  },
  'text': {
    'fontSize': 12,
    'fontWeight': 'bold',
    'color': '#fefefe',
    'paddingLeft': '120px'
  }
};
"
`;
